### PR TITLE
bpo-36367: Free buffer if realloc fails in tokenize.c

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue36367

### CODE REVIEW

**Review by matrixise — APPROVED** [review]
I had the same solution, seems to be fine for me.

**Comment by remilapeyre** [comment]
Hi @pablogsal I think there is the same issue in https://github.com/python/cpython/blob/dce242187f25b6861b08fe5d7e581efcfc49ea21/Parser/tokenizer.c#L891. Can you fix it too?

I think the cast to `char *` there is also not needed.

**Review by remilapeyre — CHANGES_REQUESTED** [review]
Hi @pablogsal I think there is the same issue in https://github.com/python/cpython/blob/dce242187f25b6861b08fe5d7e581efcfc49ea21/Parser/tokenizer.c#L891. Can you fix it too?

I think the cast to `char *` there is also not needed.

**Comment by remilapeyre** [comment]
As far as I can tell this issue is also present on https://github.com/python/cpython/blob/dce242187f25b6861b08fe5d7e581efcfc49ea21/Parser/tokenizer.c#L963

File: Parser/tokenizer.c [hunk 1]
```diff
@@ -618,7 +618,7 @@ static char *
 translate_newlines(const char *s, int exec_input, struct tok_state *tok) {
     int skip_next_lf = 0;
     size_t needed_length = strlen(s) + 2, final_length;
-    char *buf, *current;
+    char *buf, *current, *result;
```
Comment (vstinner) [inline Parser/tokenizer.c]: Can you move result declaration where it's used? Python 3.8 now allows C99 ;-) (see PEP 7)

**Comment by pablogsal** [comment]
@remilapeyre I think for https://github.com/python/cpython/blob/dce242187f25b6861b08fe5d7e581efcfc49ea21/Parser/tokenizer.c#L891 is not a problem as is handle correctly afterwards:

https://github.com/python/cpython/blob/dce242187f25b6861b08fe5d7e581efcfc49ea21/Parser/tokenizer.c#L893-L899

File: Parser/tokenizer.c (hunk 1 above)
Comment (pablogsal) [reply Parser/tokenizer.c]: Done! :)

**Review by remilapeyre — APPROVED** [review]
@pablogsal Yes it is, I think you patch is correct now.

**Pull request merged**